utils/src/main/kotlin/DiskCache.kt

```diff
@@ -80,22 +80,26 @@ class DiskCache(
      * shorten it and append the serial number.
      */
     private fun String.asKey(): String {
-        return if (length <= MAX_KEY_LENGTH - KEY_SUFFIX_LENGTH) {
+        // DiskLruCache keys must match regex [a-z0-9_-]{1,64}.
+        val mappedKey = toLowerCase().replace('.', '-').replace("[^a-z0-9_-]".toRegex(), "_")
```

Comment:
The regex should be in a field to compile it only once.

Reply:
Done.